frontend/src/individual_view.tsx

```diff
@@ -0,0 +1,215 @@
+/**
+ * @file The IndividualView function provides information about a single
+ * individual, including links to parents and progeny.
+ */
+import React from 'react'
+import { Link } from 'react-router-dom'
+import { makeStyles } from '@material-ui/core/styles'
+import { useMessageContext } from '@app/message_context'
+import { get } from '@app/communication'
+import { Individual, herdLabel, DateValue, individualLabel } from '@app/data_context_global'
+import { useDataContext } from '@app/data_context'
+import { IndividualPedigree } from '@app/individual_pedigree'
+
+const useStyles = makeStyles({
+  body: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  herdList: {
+    listStyle: 'none',
+    padding: '0 0 0 10px',
+    margin: '0 0 0 5px',
+    borderLeft: '1px solid lightgrey',
+  },
+  fillWidth: {
+    width: '100%',
+  },
+  flexColumn: {
+    minWidth: '300px',
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  sameLine: {
+    width: '100%',
+    overflow: 'hidden',
+    padding: '0 0 0 10px',
+    margin: '0 0 0 5px',
+    borderLeft: '1px solid lightgrey',
+    '& dt, dd': {
+      display: 'block',
+      float: 'left',
+    },
+    '& dt': {
+      width: '50px',
+      clear: 'both',
+    }
+  },
+  listIcon: {
+    display: 'inline-block',
+    width: '20px',
+    textAlign: 'center',
+  }
+});
+
+/**
+ * Loads information for an individual from the backend and displays it.
+ */
+export function IndividualView({id} : {id: string}) {
+  const style  = useStyles()
+  const { genebanks } = useDataContext()
+  const [individual, setIndividual] = React.useState(undefined as Individual | undefined)
+  const [children, setChildren] = React.useState([] as Individual[])
+  const {userMessage} = useMessageContext()
+
+  const activeIcon = '✅'
+  const inactiveIcon = '❌'
+  const deadIcon = '✞'
+
+  /**
+   * Fetch individual data from the backend
+   */
+  React.useEffect(() => {
+    get(`/api/individual/${id}`).then(
+      (data: Individual) => setIndividual(data),
+      error => {
+        console.error(error);
+        userMessage(error, 'error')
+      }
+    )
+  }, [id])
+
+  /**
+   * Set children
+   */
+  React.useEffect(() => {
```

Comment:
You can use the `const children = React.memo( ... )` here to avoid useState + useEffect

Reply:
good point! I'm still not quite used to `memo`!